auth-api/tests/unit/api/test_entity.py

```diff
@@ -134,6 +136,44 @@ def test_get_entity(client, jwt, session):  # pylint:disable=unused-argument
         content_type="application/json",
     )
 
+    check_auth.assert_called_once()
+    assert rv.status_code == HTTPStatus.OK
+    assert schema_utils.validate(rv.json, "business")[0]
+    dictionary = json.loads(rv.data)
+    assert dictionary["businessIdentifier"] == TestEntityInfo.entity1["businessIdentifier"]
+
+
+@mock.patch("auth_api.services.entity.check_auth")
+@mock.patch("auth_api.services.products.Product")
+def test_get_entity_as_competent_authority(
+    mock_product, check_auth, client, jwt, session
+):  # pylint:disable=unused-argument
+    """Assert that an entity can be retrieved via GET."""
+    mock_product.get_all_product_subscription.return_value = ["CA_SEARCH", "OTHER_PRODUCT"]
```

Comment:
I don't think this is a valid mock. It should be returning a list of dicts I think?